src/Sylius/Bundle/AdminBundle/shipper/OrderShipmentShipper.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Shipper;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use SM\Factory\FactoryInterface as StateMachineFactoryInterface;
+use Sylius\Component\Core\Model\ShipmentInterface;
+use Sylius\Component\Core\Repository\ShipmentRepositoryInterface;
+use Sylius\Component\Resource\Exception\UpdateHandlingException;
+use Sylius\Component\Shipping\ShipmentTransitions;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+
+final class OrderShipmentShipper implements OrderShipmentShipperInterface
+{
+    /** @var ShipmentRepositoryInterface */
+    private $shipmentRepository;
+
+    /** @var StateMachineFactoryInterface */
+    private $stateMachineFactory;
+
+    /** @var ObjectManager */
+    private $shipmentManager;
+
+    /** @var EventDispatcherInterface */
+    private $eventDispatcher;
+
+    public function __construct(
+        ShipmentRepositoryInterface $shipmentRepository,
+        StateMachineFactoryInterface $stateMachineFactory,
+        ObjectManager $shipmentManager,
+        EventDispatcherInterface $eventDispatcher
+    ) {
+        $this->shipmentRepository = $shipmentRepository;
+        $this->stateMachineFactory = $stateMachineFactory;
+        $this->shipmentManager = $shipmentManager;
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    public function ship(string $shipmentId, string $trackingCode = null): void
+    {
+        /** @var ShipmentInterface $shipment */
```

Comment:
```suggestion
        /** @var ShipmentInterface|null $shipment */
```

Also need to handle the case when `$shipment` is null.

Reply:
I'm also thinking about passing `ShipmentInterface $shipment` to this method instead of `string $shipmentId` though - this way we can eliminate this case and make it more reusable for cases when you do have an object already. I'm not sure though. I would feel safer if both this class and its interface would be marked with `@experimental` tags if `string $shipmentId` stays on this method.